src/System.Private.CoreLib/src/System/Runtime/Loader/ComponentDependencyResolver.cs

```diff
@@ -0,0 +1,282 @@
+            {
+                _assemblyPaths.Add(Path.GetFileNameWithoutExtension(assemblyPath), assemblyPath);
+            }
+
+            _nativeSearchPaths = SplitPathsList(nativeSearchPathsList);
+            _resourceSearchPaths = SplitPathsList(resourceSearchPathsList);
+
+            _componentDirectorySearchPaths = new string[1] { Path.GetDirectoryName(componentAssemblyPath) };
+        }
+
+        public string ResolveAssemblyPath(AssemblyName assemblyName)
+        {
+            // Determine if the assembly name is for a satellite assembly or not
+            // This is the same logic as in AssemblyBinder::BindByTpaList in CoreCLR
+            // - If the culture name is non-empty and it's not 'neutral' 
+            // - The culture name is the value of the AssemblyName.Culture.Name 
+            //     (CoreCLR gets this and stores it as the culture name in the internal assembly name)
+            //     AssemblyName.CultureName is just a shortcut to AssemblyName.Culture.Name.
+            if (!string.IsNullOrEmpty(assemblyName.CultureName) && 
+                !string.Equals(assemblyName.CultureName, "neutral", StringComparison.OrdinalIgnoreCase))
+            {
+                // Load satellite assembly
+                // Search resource search paths by appending the culture name and the expected assembly file name.
+                // Copies the logic in BindSatelliteResourceByResourceRoots in CoreCLR.
+                // Note that the runtime will also probe APP_PATHS the same way, but that feature is effectively 
+                // being deprecated, so we chose to not support the same behavior for components.
+                foreach (string searchPath in _resourceSearchPaths)
+                {
+                    string assemblyPath = Path.Combine(
+                        searchPath,
+                        assemblyName.CultureName,
+                        assemblyName.Name + ResourceAssemblyExtension);
+                    if (File.Exists(assemblyPath))
+                    {
+                        return assemblyPath;
+                    }
+                }
+            }
+            else
+            {
+                // Load code assembly - simply look it up in the dictionary by its simple name.
+                if (_assemblyPaths.TryGetValue(assemblyName.Name, out string assemblyPath))
+                {
+                    // Only returnd the assembly if it exists on disk - this is to make the behavior of the API
+                    // consistent. Resource and native resolutions will only return existing files
+                    // so assembly resolution should do the same.
+                    if (File.Exists(assemblyPath))
+                    {
+                        return assemblyPath;
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        public string ResolveUnmanagedDllPath(string unmanagedDllName)
```

Comment:
Please add a note here that the the CDR only resolves from the the component directory,  deps.json paths, and simple name variants. For example: It doesn't respect DllSearchPath attributes on the Module or method. Further resolution should be handled by DllImport callbacks and resolvers.

Reply:
Would it be fair to say something like:
"This only resolves the native library path by using the component directory, `.deps.json` information and simple name variants. It won't resolve libraries which would rely on the OS library resolution behavior. It should be used only as one part of a fully featured native library loading mechanism."
We should discuss the details here, as I need to understand what exactly the different options are...